tests/python/pants_test/backend/jvm/tasks/jvm_compile/java/test_zinc_compile_integration.py

```diff
@@ -254,3 +255,84 @@ def test_hermetic_binary_with_dependencies(self):
         ]:
           path = os.path.join(compile_dir, path_suffix)
           self.assertTrue(os.path.exists(path), "Want path {} to exist".format(path))
+
+  def test_hermetic_binary_cache_with_dependencies(self):
+    build_file_abs_path = os.path.join(get_buildroot(), 'examples/src/scala/org/pantsbuild/example/hello/exe/BUILD')
+
+    with open(build_file_abs_path, 'r') as f:
```

Comment:
It would be nice to extract this with-read-try-finally-write into a context manager in src/python/pants/util/contextutil.py - maybe called `with_overwriten_file_content` or similar?